lib/puma/dsl.rb

```diff
@@ -1073,7 +1088,18 @@ def process_hook(options_key, key, block, meth)
       elsif key.nil?
         @options[options_key] << block
       else
-        raise "'#{method}' key must be String or Symbol"
+        raise "'#{meth}' key must be String or Symbol"
+      end
+    end
+
+    def warn_if_in_single_mode(hook_name)
+      return if @options[:silence_fork_callback_warning]
+
+      if (@options[:workers] || 0) == 0
+        LogWriter::DEFAULT.log(
+          "Warning: You specified code to run in a `#{hook_name}` block, " \
+          "but Puma is configured to run without forking."
```

Comment:
Let's be more explicit: "but Puma is configured to run in cluster mode, so your #{hook_name} block did not run."

Reply:
thanks @nateberkopec, I have changed the log string and added test cases for it